test/functional/admin/teacher_sets_controller_test.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+module Admin
+  class TeacherSetsControllerTest < ActionController::TestCase
+
+    setup do
+      @teacher_set = teacher_sets(:teacher_set_one)
+      sign_in AdminUser.create!(email: 'admin@example.com', password: 'password')
+    end
+
+    test "test index method" do
+      get :index
+      assert_response :success
+    end
+
+    test "test show method" do
+      get :show, params: { id: @teacher_set.id }
+      assert_response :success
+    end
+
+    test "test make_available method with format js" do
+      put :make_available, params: { id: @teacher_set.id }, format: :js
+      assert_response :success
+    end
+
+    test "test make_available method" do
+      put :make_available, params: { id: @teacher_set.id }
+      assert_response :redirect
+    end
+
+    test "test make_unavailable method with format js" do
+      put :make_unavailable, params: { id: @teacher_set.id }, format: :js
+      assert_response :success
+    end
+
+    test "test make_unavailable method" do
+      put :make_unavailable, params: { id: @teacher_set.id }
+      assert_response :redirect
```

Comment:
Ok, similar question here.  It looks to me like this test and the test at line 27 are both making sure that there's _a_ redirect, but not checking where the redirect is to?

Reply:
member_action :make_unavailable, method: :put do
    teacher_set = TeacherSet.find(params[:id])
    teacher_set.availability = 'unavailable'
    teacher_set.save
    if request.format == :html
      redirect_to admin_teacher_set_path(teacher_set)
    else
      render js: "makeAvailableTeacherSet(#{teacher_set.id}, false);"
    end
  end